plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/data/json/JSONUtils.java

```diff
@@ -104,7 +104,13 @@ public static String escapeJsonString(String str) {
                 case '\f' -> result.append("\\f");
                 case '\b' -> result.append("\\b");
                 case '"', '\\', '/' -> result.append("\\").append(c);
-                default -> result.append(c);
+                default -> {
+                    if ((int) c < 32) {
```

Comment:
```suggestion
                    if (c < ' ') {
                        // escape non-displayable ASCII characters
```

Reply:
It has not so clear intention. I prefer to leave it